File: DSA/RatnaKhanna_590013141/Day_19/Question2.java
import java.util.Scanner;
class ListNode {
    int val;
    ListNode next;
    ListNode(int val) {
        this.val = val;
    }
}
public class Main {
    public static ListNode middleNode(ListNode head) {
        ListNode slow = head;
        ListNode fast = head;
        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
        }
        return slow;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        if (n == 0) {
            System.out.println("List is empty");
            return;
        }
        ListNode head = new ListNode(sc.nextInt());
        ListNode temp = head;
        for (int i = 1; i < n; i++) {
            temp.next = new ListNode(sc.nextInt());
            temp = temp.next;
        }
        ListNode middle = middleNode(head);
        System.out.println(middle.val);
        sc.close();
    }
}

File: DSA/DivinaNarula_590025969/Day_19/Question2.py
# Middle of the Linked List

class ListNode:
    def __init__(self, data):
        self.data = data
        self.next = None


def createLinkedList(arr):
    if not arr:
        return None

    head = ListNode(arr[0])
    current = head

    for value in arr[1:]:
        current.next = ListNode(value)
        current = current.next

    return head


def findMiddle(head):
    slow = head
    fast = head

    while fast and fast.next:
        slow = slow.next
        fast = fast.next.next

    return slow


def printList(head):
    current = head
    while current:
        print(current.data, end=" ")
        current = current.next
    print()


n = int(input("Enter the number of nodes: "))

print("Enter the node values:")
values = list(map(int, input().split()))

head = createLinkedList(values)

print("Linked List:")
printList(head)

middle = findMiddle(head)

print("Middle Node:", middle.data)